java/core/src/main/java/sleeper/core/util/PollWithRetries.java

```diff
@@ -77,12 +112,27 @@ public void pollUntil(String description, BooleanSupplier checkFinished) throws
         }
     }
 
+    /**
+     * Given a supplier, starts polling then return the last result from the supplier.
+     *
+     * @param  <T>                  the type of object being supplied
+     * @param  description          a short description to display when the poll fails
+     * @param  query                the supplier of {@link T}
+     * @param  condition            the exit condition applied to {@link T}
+     * @return                      the last object from the supplier
+     * @throws InterruptedException if any thread has interrupted the current thread
+     */
```

Comment:
The method and parameter names talk about a query. Please can we also use that language in the Javadoc, rather than talking about a supplier?